src/libcore/iter/range.rs

```diff
@@ -346,7 +608,7 @@ impl<A: Step> Iterator for ops::RangeInclusive<A> {
         }
         let is_iterating = self.start < self.end;
         Some(if is_iterating {
-            let n = self.start.add_one();
+            let n = Step::forward(self.start.clone(), 1);
```

Comment:
Since we checked that `start < end`, we could use `forward_unchecked` here. I'm not sure if we get any performance improvement out of it though.

Reply:
I think this applies in multiple places. I stuck to keeping the same behavior as previously and not adding `_unchecked` uses except for the previously double-checked-to-convince-LLVM-to-optimize-out-the-check case.

I think I'd prefer that if we switch more cases to using `_unchecked` to do so in a separate PR, so the performance impact of doing so can be measured separately from rewriting such a central trait.